server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -274,12 +297,16 @@ private Chromosome getNextChromosome(final List<Chromosome> chromosomes, final C
         return null;
     }
 
-    private Chromosome loadChrById(final Long referenceId, final Long chromosomeId) {
+    private Chromosome loadChrById(final Reference reference, final Long chromosomeId) {
```

Comment:
load -> fetch since we now get it from reference and only load if chromosomeId == null

Reply:
moreover how about to just get first chromosome from reference chr list?